### ISSUE TITLE
Provide feedback when a sound edit causes the sound to become too large 

### ISSUE THREAD

**Issue by ericrosenbaum** [LLK/scratch-gui#4897]
### Expected Behavior

The sound editor should provide feedback when an edit operation in the sound editor causes the sound to exceed our internal limit of 10MB. This should probably be an alert saying the sound is too large to be edited and saved by Scratch, or something similar.

### Actual Behavior

Currently, when a sound editor operation would cause the sound to exceed the 10MB limit, it fails silently (with a warning in the console: `Refusing to encode sound larger than 10000000 bytes`). This has resulted in kids reporting that the sound editor buttons do not work for some sounds.

The underlying issue is that, while you can upload compressed audio files, any edit causes the sound editor to save an uncompressed version, because we currently only have an encoder for wav files. This means the file size often increases by a factor of 10. So you can upload an mp3 that is several minutes long, but you can only edit it if it is one or two minutes long.   

### Steps to Reproduce

- Upload an mp3 file that is more than 2min or so long 
- Note that none of the effect buttons work
- Trim operations will not work unless the trimmed result is under 10MB

**Comment by ericrosenbaum** [comment]
This is related to (but distinct from) this issue, about how we need an alert when the user uploads an asset that is too large: https://github.com/LLK/scratch-gui/issues/4283

**Comment by ericrosenbaum** [comment]
cc @thisandagain

**Comment by towerofnix** [comment]
>The underlying issue is that, while you can upload compressed audio files, any edit causes the sound editor to save an uncompressed version, because we currently only have an encoder for wav files.

Yikes! :P Out of curiosity's sake, are there plans to implement an mp3 (or other compressed) format writer? I'd guess it's more of a long-term plan if it would take a big file-size/performance cost.

**Comment by ericrosenbaum** [comment]
We'd like to be saving compressed sounds- the main challenge is that while browsers support native decoding, they don't provide native encoding. Wav is uncompressed so the "encoding" is very fast. I tried out one of the available javascript mp3 encoder modules (even though it does not have a compatible license), and while it runs faster than 20x real-time, that is still pretty slow (the way the editor is set up now, re-encoding with each edit, every edit to a 3min song would cause the UI to lock up for ~9 seconds). so we'd probably want to only do the compressed encoding at the time of saving, but then auto-saving would get really slow after each sound edit, so that might not be feasible either. It definitely would require more research.

**Comment by towerofnix** [comment]
>So we'd probably want to only do the compressed encoding at the time of saving, but then auto-saving would get really slow after each sound edit, so that might not be feasible either. It definitely would require more research.

Yeah, I was just thinking that :P - while it's inevitably going to take longer to auto-save an edit during the time it's compressing, maybe you could avoid locking the browser by using a web worker? I'm not too sure how those work but it sounds like the sort of job they might apply to.
